soroban-env-host/src/events/mod.rs

```diff
@@ -123,11 +123,33 @@ impl core::fmt::Display for HostEvent {
         match &self.event.body {
             ContractEventBody::V0(ceb) => {
                 write!(f, "topics:[")?;
+
+                let mut is_fn_call = false;
                 for (i, topic) in ceb.topics.iter().enumerate() {
                     if i != 0 {
                         write!(f, ", ")?;
                     }
-                    display_scval(topic, f)?;
+
+                    // The second topic of the fn_call event is the contract id as ScBytes,
+                    // but we want to display it as a C key instead, so this block
+                    // tries to deduce if the event is the fn_call event.
+                    if i == 1 && is_fn_call {
```

Comment:
This check isn't perfect, but I don't think we need it to be.

Reply:
Yeah, I think this is good enough, given that we only format events in debug setting and the ability to imitate fn_call diagnostic doesn't seem problematic for this case.